### PR TITLE
Canceled subscription date fix

### PR DESCRIPTION
**Pull request by jamesrkiger**
## Checklist

1. [ ] If you've added code that should be tested, add tests
2. [ ] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/master/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [ ] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes

## Description

We recently had a billing unit test fail because it was testing for a subscription canceled on the last day of a 31-day month followed by a 30-day month. The billing cycle start date was being returned as August 30 instead of August 31. This PR ensures that the default plan billing cycle start date will not be earlier than the cancelation date. 
## Notes

This PR also adds the freezegun library to dev dependencies for testing. This should be useful for billing-related tests moving forward.

### CODE REVIEW

File: dependencies/pip/dev_requirements.txt [hunk 1]
```diff
@@ -685,4 +688,3 @@ yubico-client==1.13.0
 
 # The following packages are considered to be unsafe in a requirements file:
 # setuptools
-backports-zoneinfo==0.2.1; python_version < '3.9'
```
Comment (noliveleger) [inline dependencies/pip/dev_requirements.txt:688]: Do not remove `backports-zoneinfo==0.2.1; python_version < '3.9'`  until we support Python3.8

File: kobo/apps/organizations/utils.py [hunk 2]
```diff
@@ -30,10 +30,17 @@ def get_monthly_billing_dates(organization: Union[Organization, None]):
         ):
             return first_of_this_month, first_of_next_month
         
-        period_end = canceled_subscription_anchor.replace(tzinfo=pytz.UTC)
+        canceled_subscription_anchor = canceled_subscription_anchor.replace(
+            tzinfo=pytz.UTC
```
Comment (noliveleger) [inline kobo/apps/organizations/utils.py:34]: Let's use ZoneInfo and remove dependency on `pytz`

File: kobo/apps/stripe/tests/test_organization_usage.py [hunk 3]
```diff
@@ -305,6 +306,34 @@ def test_plan_canceled_last_month(self):
             == current_billing_period_end.isoformat()
         )
 
+    def test_plan_canceled_edge_date(self):
+        """
+        If a plan is canceled on the last day of a 31-day month, we want the subsequent
+        billing cycle to end on the last day of the next month, but we also need to make
+        sure the cycle starts on the cancelation date
+        """
+        cancel_date = datetime(year=2024, month=8, day=31, tzinfo=pytz.UTC)
```
Comment (noliveleger) [inline kobo/apps/stripe/tests/test_organization_usage.py:315]: Please use `zoneinfo` and not `pytz` 

```python
cancel_date = datetime(year=2024, month=8, day=31, tzinfo=ZoneInfo('UTC'))
```

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: kobo/apps/organizations/utils.py (hunk 2 above)
Comment (jamesrkiger) [reply kobo/apps/organizations/utils.py]: Done. I also removed it from kobo/apps/subsequences/actions/base.py for good measure so it isn't in our code base anymore aside from being a dependency for pandas and flower.

File: dependencies/pip/dev_requirements.txt (hunk 1 above)
Comment (jamesrkiger) [reply dependencies/pip/dev_requirements.txt:688]: Added it back (sorry, pip removed that and I should have asked about it).

Comment (jamesrkiger) [reply dependencies/pip/dev_requirements.txt]: Handled in #5125

File: kobo/apps/stripe/tests/test_organization_usage.py [hunk 4]
```diff
@@ -310,6 +312,34 @@ def test_plan_canceled_last_month(self):
...
 
+    def test_plan_canceled_edge_date(self):
+        """
+        If a plan is canceled on the last day of a 31-day month, we want the subsequent
+        billing cycle to end on the last day of the next month, but we also need to make
+        sure the cycle starts on the cancelation date
+        """
+        cancel_date = datetime(year=2024, month=8, day=31, tzinfo=ZoneInfo('UTC'))
+        with freeze_time(cancel_date.replace(day=1)):
+            subscription = generate_plan_subscription(self.organization)
```
Comment (noliveleger) [inline kobo/apps/stripe/tests/test_organization_usage.py:323]: I would use a separate variable (or maybe only a comment?) to make the purpose of `freeze_time` obvious.

```
        cancel_date = datetime(year=2024, month=8, day=31, tzinfo=ZoneInfo('UTC'))
        frozen_datetime_now = cancel_date.replace(day=1)
        with freeze_time(frozen_datetime_now):
            subscription = generate_plan_subscription(self.organization)
```

What do you think?

File: kobo/apps/stripe/tests/test_organization_usage.py [hunk 5]
```diff
@@ -310,6 +312,34 @@ def test_plan_canceled_last_month(self):
...
+
+        subscription.status = 'canceled'
+        subscription.ended_at = cancel_date
+        subscription.save()
+
+        with freeze_time(cancel_date.replace(month=9, day=1)):
+            response = self.client.get(self.detail_url)
```
Comment (noliveleger) [inline kobo/apps/stripe/tests/test_organization_usage.py:330]: Same comment ;-)

File: kobo/apps/stripe/tests/test_organization_usage.py (hunk 4 above)
Comment (jamesrkiger) [reply kobo/apps/stripe/tests/test_organization_usage.py]: Yeah, I think that's a good idea. I was trying to be economical with line length, but it makes more sense to make the code explain itself.

**Pull request merged**